server/controllers/webhooks.js

```diff
@@ -73,3 +75,61 @@ module.exports.fetchGithubActivities = async (req, res) => {
     return res.status(500).write("error: true");
   }
 };
+
+module.exports.createGithubWebhook = async (req, res) => {
+  try {
+    if (req.cookies["gh_private_repo_token"]) {
+      const secret_token =
+        Math.random().toString(36).slice(2) +
+        Math.random().toString(36).toUpperCase().slice(2);
+      const headers = {
+        accept: "application/vnd.github.v3+json",
+        Authorization: `token ${req.cookies["gh_private_repo_token"]}`,
+      };
+
+      const config = {
+        //Change this URL to the server's public payload URL
+        url: "http://8e46119e3447.ngrok.io/webhooks/github",
```

Comment:
The domain name must be ${rc4gitApiDomain}, and shouldn't be hardcoded

Reply:
I haven't used `rc4gitApiDomain` here since the payload slug should be accessible on the public internet. Because we're running everything locally for now (and localhost isn't publicly accessible) therefore this shouldn't be rc4gitApiDomain **for now**.
Once we're live publicly this should be changed. The comment correspondingly acts as a reminder.

Reply:
No actually, we need to set our constant rc4gitApiDomain with the ngrok domain which eventually points to our local domain.
This will prevent hardcoding and solve the problem